evalml/pipelines/components/estimators/regressors/prophet_regressor.py

```diff
@@ -0,0 +1,155 @@
+import copy
+
+import numpy as np
+import pandas as pd
+from skopt.space import Real
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import import_or_raise, infer_feature_types
+from evalml.utils.gen_utils import classproperty, suppress_stdout_stderr
+
+
+class ProphetRegressor(Estimator):
+    """
+    Prophet is a procedure for forecasting time series data based on an additive model where non-linear trends are fit with yearly, weekly, and daily seasonality, plus holiday effects.
+    It works best with time series that have strong seasonal effects and several seasons of historical data. Prophet is robust to missing data and shifts in the trend, and typically handles outliers well.
+
+    More information here: https://facebook.github.io/prophet/
+
+    """
+
+    name = "Prophet Regressor"
+    hyperparameter_ranges = {
+        "changepoint_prior_scale": Real(0.001, 0.5),
+        "seasonality_prior_scale": Real(0.01, 10),
+        "holidays_prior_scale": Real(0.01, 10),
+        "seasonality_mode": ["additive", "multiplicative"],
+    }
+    model_family = ModelFamily.PROPHET
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(
+        self,
+        date_column="ds",
+        changepoint_prior_scale=0.05,
+        seasonality_prior_scale=10,
+        holidays_prior_scale=10,
+        seasonality_mode="additive",
+        random_seed=0,
+        stan_backend="CMDSTANPY",
+        **kwargs
+    ):
+        self.date_column = date_column
+
+        parameters = {
+            "changepoint_prior_scale": changepoint_prior_scale,
+            "seasonality_prior_scale": seasonality_prior_scale,
+            "holidays_prior_scale": holidays_prior_scale,
+            "seasonality_mode": seasonality_mode,
+            "stan_backend": stan_backend,
+        }
+
+        parameters.update(kwargs)
+
+        c_error_msg = (
+            "cmdstanpy is not installed. Please install using `pip install cmdstanpy`."
+        )
+        _ = import_or_raise("cmdstanpy", error_msg=c_error_msg)
+
+        p_error_msg = (
+            "prophet is not installed. Please install using `pip install prophet`."
+        )
+        prophet = import_or_raise("prophet", error_msg=p_error_msg)
+
+        prophet_regressor = prophet.Prophet(**parameters)
```

Comment:
If a user does `pip install evalml` but they didn't go through the steps to install prophet, they would get a weird error about the model not being compiled right? 

Can we add some more statements like `import_or_raise` to make sure prophet was installed with the right backend? We may have to check that the `prophet_model.bin` file is present? 


I think the one flaw with our approach for setting up the installation right now is that it lets users get into this state by "default" if they do `pip install evalml`. I think this may be a good reason to do use the `extras_require` pattern like woodwork [does](https://github.com/alteryx/woodwork/blob/main/setup.py#L9).

FYI @dsherry

Reply:
> If a user does pip install evalml but they didn't go through the steps to install prophet, they would get a weird error about the model not being compiled right?
> 
> Can we add some more statements like import_or_raise to make sure prophet was installed with the right backend? We may have to check that the prophet_model.bin file is present?

Our current install pattern is:
* Running `pip install evalml` installs everything from `core-requirements.txt` and the optional reqs in `requirements.txt`.
* We also support users manually installing only the stuff listed in `core-requirements.txt`. In which case, automl should not try to run xgboost/lightgbm/prophet, and users will get a clear error message if they try to run those estimators by hand.

I don't think this PR means we need to change that pattern. Prophet is another estimator like xgboost/lightgbm which we list as an "optional" dependency. Right?

Reply:
@dsherry I think the one difference with prophet and xgboost is that if users do `pip install evalml` and then train/score with an xgboost estimator, nothing will error.

But if they do `pip install evalml` and then train/score with prophet, they will get an error because `pip install evalml` will not set up the `cmdstan` backend. To make it confusing, prophet is installed! So the `import_or_raise("prophet")` calls will not catch this. 

Given that prophet requires an extra step to properly set up the stan backend, my hope with using `extras_require` is that it won't install prophet by default and kind of force users to read our docs to find out how they should properly set up prophet. I think this is worth doing even if we get our `stan_ext` package off the ground, because users will still have to set up some env variables to properly tell prophet where the stan binaries are.

Reply:
Ah!!! Great point.

In that case, yes, I support this. Let's have clear install documentation for how to get prophet to work, clean behavior in automl and in the prophet component for our default installation (`pip install evalml`), and then also an extras install path to get prophet set up correctly. Right?

FWIW I don't think that blocks merging this PR, but I agree that we should get a fix merged.

Reply:
Sounds good! Agree we can do it in a follow up issue.